lib/list.gi

```diff
@@ -3371,7 +3371,11 @@ InstallOtherMethod( \*,
           if depthDiff < -1 or depthDiff > 1 then
               return PROD_LISTS_SPECIAL(left, right, depth1, depth2 );
           else
-              return PROD_LIST_LIST_DEFAULT( left, right, depthDiff );
+              if IsEmpty( left ) then
+                  return [];
```

Comment:
I just checked, and in the master branch, this function is identical. So this seems like a cure for the symptom, not the actual cause.